src/shared/components/DateWidget.tsx

```diff
@@ -0,0 +1,26 @@
+import { FC } from "react";
+import { TextFieldProps } from "@material-ui/core";
+import MuiDatePicker from "@material-ui/lab/DatePicker";
+import AdapterDateFns from "@material-ui/lab/AdapterDateFns";
+import LocalizationProvider from "@material-ui/lab/LocalizationProvider";
+import { format, isValid } from "date-fns";
+import { WidgetProps } from "@rjsf/core";
+import { TextWidget } from "./TextWidget";
+
+// TODO: Fix typings for TextWidget
+export const DateWidget: FC<WidgetProps> = (props: any) => {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <MuiDatePicker
+        label={props.label || props.schema.title}
+        value={props.value ?? null}
+        onChange={(newValue) => {
+          if (isValid(newValue)) props.onChange(format(newValue, "dd/MM/yyyy"));
+        }}
+        renderInput={(params: TextFieldProps) => (
+          <TextWidget {...params} {...props} />
+        )}
+      />
+    </LocalizationProvider>
+  );
```

Comment:
so I can use a text widget and pass 'required' which does the 're-focusing' of the input when you try submit and it's empty.. also has null as the default value which is .. nice.